src/mon/Elector.cc

```diff
@@ -491,6 +491,15 @@ void Elector::send_peer_ping(int peer, const utime_t *n)
 void Elector::ping_check(int peer)
 {
   dout(20) << __func__ << " to peer " << peer << dendl;
+
+  if (peer >= mon->monmap->ranks.size()) {
+    // Monitor no longer exists in the monmap,
+    // therefore, we shouldn't ping this monitor
+    // since we cannot lookup the address!
+    dout(20) << __func__ << "peer >= ranks_size" << dendl;
```

Comment:
nit: not sure if logging the actual ranks.size() and peer as part of this line could be useful in some debugging scenario

Reply:
Yes that will be useful, thank you for the suggestion I'll file another PR addressing that